### ISSUE TITLE
A way to regulate throttle on propellers, thrusters, jet engines, motors etc.

### ISSUE THREAD

**Issue by YourLocalPlonker** [studio-minus/ppg-bugs-and-requests#2088]
**Feature description**
Instead of engines just have an on and off and going from 0 throttle to max throttle, there should be an ability to regulate how much thrust is being produced. When an engine is off it produces no throttle, when it is on it produces very little throttle (except for the thrusters as those are technically liquid rocket engines) and there should be a possibility to bind keys with throttle up.

**Feature purpose**
It would make aircraft more realistic as no aircraft flies at full throttle at all times. Would also eliminate the need of several propellers in helicopters (instead of 3 propellers, one for hovering, one for climbing and one for descending) as now the amount of throttle generated can be adjusted with the single propeller.

**Additional context**
By binding keys to adjusting the throttle I mean it should be in the UI when right-clicking the engine. Also this feature isn't like powering an engine with electricity as that will not allow you to make precise adjustments but just a rough addition of thrust

**Comment by 69XAnonymousX69** [comment]
I was just going to write this, yes it totally needs it.
also, the afterburners should be optional, would be cool if you could control it with R, G and blue wires (r for thruttle up, g for down, b for afterburner toggle) also the engine can't just start instantly, there needs to be a one second delay before it could start producing thrust.

**Comment by YourLocalPlonker** [comment]
Well I think the wires is not such a good idea, because green activates the engine and while red and blue would be good for throttle control, there wouldn't be a 4th wire to put the throttle into the afterburner detent. So I am still gonna stick with the UI idea

**Comment by DindinYT37** [comment]
Yeah, but then there'd be no way to automate it with machinery, which is one of the main aspects of People Playground.

*Potentially a new type of wire could solve this, which would replace the functions of the red and blue propagation wires of the other machinery as well. Though, the red and blue propagation wires should keep their functionality on the older machinery as otherwise it would break a lot of things. But I generally doubt it's worth adding a new kind of wire for something like this, unless there are more features that could use this - it wouldn't really be worth.*

**Comment by 69XAnonymousX69** [comment]
> Well I think the wires is not such a good idea, because green activates the engine and while red and blue would be good for throttle control, there wouldn't be a 4th wire to put the throttle into the afterburner detent. So I am still gonna stick with the UI idea

well the afterburner could activate at 100% throttle.

**Comment by YourLocalPlonker** [comment]
> > Well I think the wires is not such a good idea, because green activates the engine and while red and blue would be good for throttle control, there wouldn't be a 4th wire to put the throttle into the afterburner detent. So I am still gonna stick with the UI idea
> 
> well the afterburner could activate at 100% throttle.

That would be already pretty good but in real jets afterburner is not full throttle but a mode in which fuel is injected into directly into the afterburning chamber and also the amount of fuel Injected can be regulated. It is like >80% when afterburner is turned on. But some jets like the F/A-18 turn the afterburner on automatically when the throttle is put above 80%. So an extra wire isn't very necessary to turn on the afterburner.

**Comment by 69XAnonymousX69** [comment]
> > > Well I think the wires is not such a good idea, because green activates the engine and while red and blue would be good for throttle control, there wouldn't be a 4th wire to put the throttle into the afterburner detent. So I am still gonna stick with the UI idea
> > 
> > 
> > well the afterburner could activate at 100% throttle.
> 
> That would be already pretty good but in real jets afterburner is not full throttle but a mode in which fuel is injected into directly into the afterburning chamber and also the amount of fuel Injected can be regulated. It is like >80% when afterburner is turned on. But some jets like the F/A-18 turn the afterburner on automatically when the throttle is put above 80%. So an extra wire isn't very necessary to turn on the afterburner.

Exactly
